src/java/org/apache/cassandra/service/ActiveRepairService.java

```diff
@@ -525,6 +525,20 @@ public synchronized ParentRepairSession removeParentRepairSession(UUID parentSes
         return parentRepairSessions.remove(parentSessionId);
     }
 
+    public synchronized ParentRepairSession removeParentRepairSessionIfPresent(UUID parentSessionId)
```

Comment:
this method looks very similar to `removeParentRepairSession`.. should we refactor it?

alternatively, we can change `removeParentRepairSession` to gracefully handle non-existing parent session.. there is no point throw NPE if parent session is already gone..

Reply:
they are the same minus the "don't fail if not present thing". The code that called `removeParentRepairSession` assumes the session exists so would be an error it it doesn't, the new code which calls `removeParentRepairSessionIfPresent` calls it during cases were the session may not be setup yet, so may see a null.

I could refactor so the `removeParentRepairSession` calls this one and still fails, but I didn't want to change the semantics of that method.

Reply:
got rid of this method and made `removeParentRepairSession` handle the case the map doesn't know of the session.